Automattic/wp-calypso
# DM Hackathon- Add a daily prompt modal to editor.

Related to # DM Hackathon Project

https://github.com/Automattic/wp-calypso/pull/84240 adds query args to trigger this within the blogger onboarding flow.

## Proposed Changes
DM HACKATHON PROJECT

* adds a modal to allow users to select a daily prompt. adds the daily prompt block when selected. modal opens when `new_prompt` query arg is present.
* adds tracking events for `calypso_editor_writing_prompts_modal_viewed`, `calypso_editor_writing_prompts_modal_prompt_selected` (with a `promptId` prop), and `calypso_editor_writing_prompts_modal_closed`.

![prompt-modal-2-small](https://github.com/Automattic/wp-calypso/assets/28742426/311aafec-db77-44ba-9bbd-2da801797b3c)


## Testing Instructions



Setup:
* Sandbox a test site, public-api, and widgets.wp.com
* build and sync wpcom-block-editor from this branch ( `yarn dev --sync` from apps/wpcom-block-editor)
* build and sync editing-toolkit from this branch (`yarn dev --sync` from apps/editing-toolkit)
* build calypso from this branch

Test the Modal:
* go to the path `*/post/{siteSlug}/?new_prompt=true` to open the editor with this arg present and the modal should appear to browse and select prompts.
* test tracking events for the modal - `calypso_editor_writing_prompts_modal_viewed`, `calypso_editor_writing_prompts_modal_prompt_selected` (with a `promptId` prop), and `calypso_editor_writing_prompts_modal_closed` - and verify they fire when expected.

Test the writing prompts card on My Home:
* go to my home on local calypso for a blogging site where the daily prompts card is available.
* select a writing prompt and select "Post Answer"
* verify that the editor loads with the writing prompt block for this prompt in place.

Smoke check entering the editor from other sources and without these query params.

## Pre-merge Checklist
